src/FoxyClient.php

```diff
@@ -0,0 +1,637 @@
+<?php
+
+namespace Foxy\FoxyClient;
+
+/*
+
+config:
+    use_sandbox (boolean): set to true to connect to the API sandbox for testing.
+    access_token (string): pass in your properly scoped OAuth access_token to be used as a Authentication Bearer HTTP header.
+
+Examples:
+
+$psr17Factory = new \Nyholm\Psr7\Factory\Psr17Factory();
+$psr18Client = new \Buzz\Client\Curl($psr17Factory);
+
+//Get Homepage
+$fc = new FoxyClient($psr17Factory, $psr18Client);
+$result = $fc->get();
+die("<pre>" . print_r($result, 1) . "</pre>");
+
+//Get Authenticated Store
+$fc = new FoxyClient($psr17Factory, $psr18Client, [
+        'client_id' => $client_id,
+        'client_secret' => $client_secret,
+        'refresh_token' => $refresh_token,
+        'access_token' => $access_token, // optional
+        'access_token_expires' => $access_token_expires // optional
+    ]
+);
+$fc->get();
+$result = $fc->get($fc->getLink("fx:store"));
+die("<pre>" . print_r($result, 1) . "</pre>");
+
+Credits: David Hollander of https://foxytools.com/ got this project going and is awesome.
+*/
+
+use Exception;
+use Foxy\FoxyClient\Exceptions\JsonException;
+use Foxy\FoxyClient\Exceptions\ResponseException;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class FoxyClient
+{
+    const PRODUCTION_API_HOME = 'https://api.foxycart.com';
+    const SANDBOX_API_HOME = 'https://api-sandbox.foxycart.com';
+    const LINK_RELATIONSHIPS_BASE_URI = 'https://api.foxycart.com/rels/';
+    const PRODUCTION_AUTHORIZATION_ENDPOINT = 'https://my.foxycart.com/authorize';
+    const SANDBOX_AUTHORIZATION_ENDPOINT = 'https://my-sandbox.foxycart.com/authorize';
+    const DEFAULT_ACCEPT_CONTENT_TYPE = 'application/hal+json';
+    private static array $valid_config_options = [
+        'api_home',
+        'authorization_endpoint',
+        'use_sandbox',
+        'access_token',
+        'access_token_expires',
+        'refresh_token',
+        'client_id',
+        'client_secret',
+        'handle_exceptions'
+    ];
+    /**
+     * OAuth Access Token (can have various scopes such as client_full_access, user_full_access, store_full_access)
+     */
+    private string $access_token = '';
+    /**
+     * The timestamp for when the access token expires
+     */
+    private int $access_token_expires = 0;
+    /**
+     * OAuth Refresh token used to obtain new Access Token
+     */
+    private string $refresh_token = '';
+    /**
+     * OAuth endpoint for getting new Access Token
+     */
+    private ?string $oauth_token_endpoint = null;
```

Comment:
This doesn't feel right. Specifically initializing a property as string and immediately assigning its value as null. Sure it won't break anything but I guess it will make Programming Gods sad? (sorry if am missing some big picture/context here)
@brettflorio

Reply:
How so? It's defined as nullish string, before it was just string inited with an empty string value and the correct value is added from the API response: https://github.com/FoxyCart/foxyclient-php/blob/master/src/Foxy/FoxyClient/FoxyClient.php#L511

(I didn't make these decisions, just cleaned and added correct types)

Reply:
Hey @theboxer Thanks for getting back. That's okay. But just for the clarification, null and string are two different data types in PHP. It's just that PHP is kind enough not to complain about this.

Reply:
Hi @Omeryl thanks for the knowledge and correcting me. I kind of mixed it with how dangling **?** gets treated in typescript/JavaScript.